contracts/src/v0.8/dev/vrf/VRFV2PlusWrapper.sol

```diff
@@ -67,33 +51,53 @@ contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsume
   // charges.
   uint32 private s_fulfillmentFlatFeeNativePPM;
 
+  LinkTokenInterface public s_link;
+
   // Other configuration
 
   // s_wrapperGasOverhead reflects the gas overhead of the wrapper's fulfillRandomWords
   // function. The cost for this gas is passed to the user.
   uint32 private s_wrapperGasOverhead;
 
+  // Configuration fetched from VRFCoordinatorV2
+
+  /// @dev this is the size of a VRF v2 fulfillment's calldata abi-encoded in bytes.
+  /// @dev proofSize = 13 words = 13 * 256 = 3328 bits
+  /// @dev commitmentSize = 5 words = 5 * 256 = 1280 bits
+  /// @dev dataSize = proofSize + commitmentSize = 4608 bits
+  /// @dev selector = 32 bits
+  /// @dev total data size = 4608 bits + 32 bits = 4640 bits = 580 bytes
+  uint32 public s_fulfillmentTxSizeBytes = 580;
+
   // s_coordinatorGasOverhead reflects the gas overhead of the coordinator's fulfillRandomWords
   // function. The cost for this gas is billed to the subscription, and must therefor be included
   // in the pricing for wrapped requests. This includes the gas costs of proof verification and
   // payment calculation in the coordinator.
   uint32 private s_coordinatorGasOverhead;
 
+  AggregatorV3Interface public s_linkNativeFeed;
+
+  // s_configured tracks whether this contract has been configured. If not configured, randomness
+  // requests cannot be made.
+  bool public s_configured;
+
+  // s_disabled disables the contract when true. When disabled, new VRF requests cannot be made
+  // but existing ones can still be fulfilled.
+  bool public s_disabled;
+
   // s_wrapperPremiumPercentage is the premium ratio in percentage. For example, a value of 0
   // indicates no premium. A value of 15 indicates a 15 percent premium.
   uint8 private s_wrapperPremiumPercentage;
 
-  // s_keyHash is the key hash to use when requesting randomness. Fees are paid based on current gas
-  // fees, so this should be set to the highest gas lane on the network.
-  bytes32 s_keyHash;
-
   // s_maxNumWords is the max number of words that can be requested in a single wrapped VRF request.
   uint8 s_maxNumWords;
 
+  ExtendedVRFCoordinatorV2PlusInterface public immutable COORDINATOR;
+
   struct Callback {
     address callbackAddress;
     uint32 callbackGasLimit;
-    uint256 requestGasPrice;
+    uint64 requestGasPrice;
```

Comment:
Minor: worth a comment here around the logic of this change. Mentioning that 18 ether would be the max gas price we can process would be useful, as well as mentioning that it's unlikely to be used on most chains.

Reply:
Added the comment